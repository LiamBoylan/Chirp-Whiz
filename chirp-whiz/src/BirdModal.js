import React from 'react';
import birdList from './birdList';
import AudioButton from './quizComponents/AudioButton';
import 'bootstrap/dist/css/bootstrap.min.css';
import Modal from 'react-bootstrap/Modal';
import Container from 'react-bootstrap/Container';
import Col from 'react-bootstrap/Col';
import Row from 'react-bootstrap/Row';
import Button from 'react-bootstrap/Button';
import Image from 'react-bootstrap/Image';

function BirdModal(props) {
    return(
        <Modal
            show={props.show}
            onHide={props.onHide}
            size="lg"
            centered
        >
            <Modal.Header closeButton>
                <Modal.Title>{birdList[props.birdNum].name}</Modal.Title>
            </Modal.Header>
            <Modal.Body>
                <Container>
                    <Row className="show-grid">
                        <Col>
                <Image src={birdList[props.birdNum].image[0]} rounded style={{height: '360px'}} />
                </Col>
                </Row>
                <Row className="show-grid">
                    <Col>
                <AudioButton sound={birdList[props.birdNum].sound}/>
                </Col>
                </Row>
                </Container>
            </Modal.Body>
            <Modal.Footer>
                <Button variant="secondary" onClick={props.onHide}>Close</Button>
            </Modal.Footer>
        </Modal>
    );
}

export default BirdModal;