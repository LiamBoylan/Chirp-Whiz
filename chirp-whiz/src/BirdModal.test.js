import React from 'react';
import BirdModal from './BirdModal';
import '@testing-library/jest-dom/extend-expect'
import { render, fireEvent, waitForElement } from '@testing-library/react'

const BModalTst = require('./BirdModal.js');

// Integration tests

it('BirdModal snapshot', () => {
    const container = render(<BirdModal
        show={true}
        onHide={() => {}}
        birdNum={0} />);
    expect(container.firstChild).toMatchSnapshot();
});