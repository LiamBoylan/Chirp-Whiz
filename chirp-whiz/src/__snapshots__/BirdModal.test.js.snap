// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BirdModal snapshot 1`] = `undefined`;
